### DISCUSSION TITLE
Difference in EE estimates between Ecopath and Rpath

### DISCUSSION THREAD

**Discussion by max-cosme** [NOAA-EDAB/Rpath#131] [General]
Hi !

I tried to estimate parameters from a same set of basic inputs and diet compositions in Ecopath and Rpath. 

- Basic inputs : [toy_basic_input.csv](https://github.com/NOAA-EDAB/Rpath/files/13647044/toy_basic_input.csv)
- Diet compositions : [toy_diet_composition.csv](https://github.com/NOAA-EDAB/Rpath/files/13647046/toy_diet_composition.csv)

Everything works fine, except for Ecotrophic Efficiency (EE). 

I've tried diving into the `rpath()` function but I can't see where the problem is coming from.

---
**Rpath** results
| Group | type | TL | Biomass | PB | QB | EE | GE |
|-------|-----|----|---------|----|----|----|--------------|
| Predator | 0  | 3.173469 | 1  | 2  | 10 | 0 | 0.2000000 | 0 |
| H1       | 0  | 2.224490 | 1  | 10 | 19 | 0 | 0.5263158 | 0 |
| H2       | 0  | 2.122449 | 1  | 10 | 20 | 0 | 0.5000000 | 0 |
| Plant    | 1  | 1.000000 | 10 | 50 | 0  | 1 | 0.0000000 | 0 |
| Detritus | 2  | 1.000000 | 1  | 0  | 0  | 0 | 0.0000000 | 0 |
| Fishers  | 3  | 1.000000 | 0  | 0  | 0  | 0 | 0.0000000 | 0 |

**Ecopath** results
| Group | type | TL | Biomass | PB | QB | EE | GE |
|-------|-----|----|---------|----|----|----|--|
| Predator | 0  | 3.173 | 1  | 2  | 10 | 0 | 0.2 |
| H1       | 0  | 2.224 | 1  | 10 | 19 | 0.7 | 0.526 | 
| H2       | 0  | 2.122 | 1  | 10 | 20 | 0.88 | 0.5 | 
| Plant    | 1  | 1 | 10 | 50 | 0  | 1 |  | 
| Detritus | 2  | 1 | 1  | 0  | 0  | 0 |  | 
| Fishers  | 3  | 1 | 0  | 0  | 0  | 0 |  |

**Comment by Clara-yong** [comment]
Hi!  

I am new to RPath and am facing a similar issue except that my Rpath results returned "0" EE for all groups. It also gave very different results when I ran Rsim. 

**Basic Parameters:** 
[Model BP.csv](https://github.com/NOAA-EDAB/Rpath/files/14884381/Model.BP.csv)

**Diet composition:** 
[Model DC.csv](https://github.com/NOAA-EDAB/Rpath/files/14884398/Model.DC.csv)

**R code:** 
[Rpath EE troubleshoot.txt](https://github.com/NOAA-EDAB/Rpath/files/14884788/Rpath.EE.troubleshoot.txt)






**Ecopath with Ecosim output** 

<html xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=PowerPoint.Slide>
<meta name=Generator content="Microsoft PowerPoint 15">
<style>

</style>
</head>

<body>



Group name | Trophic level | Hab area (proportion) | Biomass in habitat area (t/km²) | Biomass (t/km²) | Production / biomass (/year) | Consumption / biomass (/year) | Ecotrophic Efficiency | Production / consumption (/year)
-- | -- | -- | -- | -- | -- | -- | -- | --
C2 | 4 | 1 | 0,5 | 0,5 | 1 | 4 | 0 | 0,25
C1 | 3 | 1 | 1 | 1 | 5 | 12,5 | 0,4 | 0,4
Zooplankton | 2 | 1 | 2 | 2 | 10 | 50 | 0,625 | 0,2
Phytoplankton | 1 | 1 | 5 | 5 | 100 |   | 0,2 |  
Detritus | 1 | 1 | 10 | 10 |   |   | 0 |  




</body>

</html>




**RPath output** 

<html xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=PowerPoint.Slide>
<meta name=Generator content="Microsoft PowerPoint 15">
<style>

</style>
</head>

<body>



Group | type | TL | Biomass | PB | QB | EE | GE | Removals
-- | -- | -- | -- | -- | -- | -- | -- | --
Gastropod | 0 | 4 | 0.5 | 1 | 4 | 0 | 0.25 | 0
Coral | 0 | 3 | 1 | 5 | 12.5 | 0 | 0.4 | 0
Zooplankton | 0 | 2 | 2 | 10 | 50 | 0 | 0.2 | 0
Phytoplankton | 1 | 1 | 5 | 100 | 0 | 0 | 0 | 0
Detritus | 2 | 1 | 10 | 0 | 0 | 0 | 0 | 0
Fleet | 3 | 1 | 0 | 0 | 0 | 0 | 0 | 0




</body>

</html>


**Ecosim output**
[Basic model_troubleshooting EE_EwE.csv](https://github.com/NOAA-EDAB/Rpath/files/14884815/Basic.model_troubleshooting.EE_EwE.csv)


--summarized--

<html xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=PowerPoint.Slide>
<meta name=Generator content="Microsoft PowerPoint 15">
<style>

</style>
</head>

<body>



Pane | Cumulative biomass |   |   |   |  
-- | -- | -- | -- | -- | --
Data | 1: C2 | 2: C1 | 3: Zooplankton | 4: Phytoplankton | 5: Detritus |  
0 | 0.5 | 1 | 2 | 5 | 10 |  
1.083333 | 0.500005 | 1.000028 | 2.000021 | 4.99997 | 9.999887 |  
1.166667 | 0.500005 | 1.00003 | 2.000115 | 5.000002 | 9.99988 |  
… |   |   |   |   |   |  
101 | 0.5 | 1.000001 | 2 | 5 | 9.999868 |  




</body>

</html>





**Rsim output**

<html xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=PowerPoint.Slide>
<meta name=Generator content="Microsoft PowerPoint 15">
<style>

</style>
</head>

<body>



Group | StartBio | EndBio | BioES | StartCatch | EndCatch | CatchES
-- | -- | -- | -- | -- | -- | --
Outside | 1 | 1 | 1 | 0 | 0 | NA
C2 | 0.5 | 5.00E-16 | 1.00E-15 | 0 | 0 | NA
C1 | 1 | 0.0114 | 0.0114 | 0 | 0 | NA
Zooplankton | 2 | 1.0109 | 0.50545 | 0 | 0 | NA
Phytoplankton | 5 | 3.815773 | 0.763155 | 0 | 0 | NA
Detritus | 10 | 40225.39 | 4022.539 | 0 | 0 | NA
Fleet | 0 | 0 | NA | 0 | 0 | NA




</body>

</html>

**Comment by ghost** [comment]
> Hi!
> 
> I am new to RPath and am facing a similar issue except that my Rpath results returned "0" EE for all groups. It also gave very different results when I ran Rsim.
> 
> **Basic Parameters:** [Model BP.csv](https://github.com/NOAA-EDAB/Rpath/files/14884381/Model.BP.csv)
> 
> **Diet composition:** [Model DC.csv](https://github.com/NOAA-EDAB/Rpath/files/14884398/Model.DC.csv)
> 
> **R code:** [Rpath EE troubleshoot.txt](https://github.com/NOAA-EDAB/Rpath/files/14884788/Rpath.EE.troubleshoot.txt)
> 
> **Ecopath with Ecosim output**
> 
> <style> </style>
> Group name	Trophic level	Hab area (proportion)	Biomass in habitat area (t/km²)	Biomass (t/km²)	Production / biomass (/year)	Consumption / biomass (/year)	Ecotrophic Efficiency	Production / consumption (/year)
> C2	4	1	0,5	0,5	1	4	0	0,25
> C1	3	1	1	1	5	12,5	0,4	0,4
> Zooplankton	2	1	2	2	10	50	0,625	0,2
> Phytoplankton	1	1	5	5	100	 	0,2	 
> Detritus	1	1	10	10	 	 	0	 
> **RPath output**
> 
> <style> </style>
> Group	type	TL	Biomass	PB	QB	EE	GE	Removals
> Gastropod	0	4	0.5	1	4	0	0.25	0
> Coral	0	3	1	5	12.5	0	0.4	0
> Zooplankton	0	2	2	10	50	0	0.2	0
> Phytoplankton	1	1	5	100	0	0	0	0
> Detritus	2	1	10	0	0	0	0	0
> Fleet	3	1	0	0	0	0	0	0
> **Ecosim output** [Basic model_troubleshooting EE_EwE.csv](https://github.com/NOAA-EDAB/Rpath/files/14884815/Basic.model_troubleshooting.EE_EwE.csv)
> 
> --summarized--
> 
> <style> </style>
> Pane	Cumulative biomass	 	 	 	 
> Data	1: C2	2: C1	3: Zooplankton	4: Phytoplankton	5: Detritus
> 0	0.5	1	2	5	10
> 1.083333	0.500005	1.000028	2.000021	4.99997	9.999887
> 1.166667	0.500005	1.00003	2.000115	5.000002	9.99988
> …	 	 	 	 	 
> 101	0.5	1.000001	2	5	9.999868
> **Rsim output**
> 
> <style> </style>
> Group	StartBio	EndBio	BioES	StartCatch	EndCatch	CatchES
> Outside	1	1	1	0	0	NA
> C2	0.5	5.00E-16	1.00E-15	0	0	NA
> C1	1	0.0114	0.0114	0	0	NA
> Zooplankton	2	1.0109	0.50545	0	0	NA
> Phytoplankton	5	3.815773	0.763155	0	0	NA
> Detritus	10	40225.39	4022.539	0	0	NA
> Fleet	0	0	NA	0	0	NA

I got this problem too!

**Comment by max-cosme** [comment]
And I re-checked : the bug is still here with my example... 

Hope this won't stuck you in your work ! :/

**Comment by kaydin** [comment]
Closing discussion as this has moved to an issue.
